datadog/fwprovider/data_source_datadog_application_key.go

```diff
@@ -0,0 +1,119 @@
+package fwprovider
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var _ datasource.DataSource = &applicationKeyDataSource{}
+
+type applicationKeyDataSourceModel struct {
+	Id   types.String `tfsdk:"id"`
+	Name types.String `tfsdk:"name"`
+	Key  types.String `tfsdk:"key"`
+}
+
+type applicationKeyDataSource struct {
+	Api  *datadogV2.KeyManagementApi
+	Auth context.Context
+}
+
+func NewApplicationKeyDataSource() datasource.DataSource {
+	return &applicationKeyDataSource{}
+}
+
+// Metadata implements datasource.DataSource.
+func (d *applicationKeyDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "application_key"
+}
+
+// Schema implements datasource.DataSource.
+func (d *applicationKeyDataSource) Schema(_ context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Use this data source to retrieve information about an existing application key.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Id for Application Key.",
+				Optional:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "Name for Application Key.",
+				Optional:    true,
+			},
+			"key": schema.StringAttribute{
+				Description: "The value of the Application Key.",
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func (r *applicationKeyDataSource) Configure(_ context.Context, request datasource.ConfigureRequest, response *datasource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetKeyManagementApiV2()
+	r.Auth = providerData.Auth
+}
+
+// Read implements datasource.DataSource.
+func (d *applicationKeyDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state applicationKeyDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if !state.Id.IsNull() {
+		ddResp, _, err := d.Api.GetApplicationKey(d.Auth, state.Id.ValueString())
```

Comment:
Looks like you are using the wrong operation. Correct one is `GetCurrentUserApplicationKey`. The `GetApplicationKey` doesn't return the actual key. 

Could you also add an integration only test for this case and assert the app key is not empty in state.